packages/common/src/util.ts

```diff
@@ -21,3 +28,36 @@ export const waitUntilCommitted = async (rpc: RPC, txHash: string, timeout = 120
     await scheduler.wait(1000)
   }
 }
+
+export const createPath = (path: string) => {
+  if (!fs.existsSync(path)) {
+    fs.mkdirSync(path, { recursive: true })
+  }
+  return path
+}
+
+export const cachePath = (...paths: string[]) => createPath(path.resolve(PATH.cache, ...paths))
+
+export const configPath = (...paths: string[]) => createPath(path.resolve(PATH.config, ...paths))
+
+export async function downloadFile(url: string | URL | UrlObject, filePath: string) {
+  try {
+    const { body } = await undici.request(url, { method: 'GET', maxRedirections: MAX_REDIRECTS })
+
+    const fileStream = fs.createWriteStream(filePath)
+
+    await new Promise((res, rej) => {
+      body
+        .pipe(fileStream)
+        .on('finish', () => {
+          fileStream.close()
+          res(0)
+        })
+        .on('error', (error) => {
+          rej(error)
+        })
+    })
+  } catch (error) {
+    console.error('Error downloading file:', error)
+  }
```

Comment:
It can be simplified as

```javascript
import { pipeline } from 'node:stream/promises'
import fs from 'node:fs'

pipeline(
  await fetch(URL).then(res => res.body),
  fs.createReadableStream(filePath),
).catch(err => console.error('Error on downloading file: ', error)
```

Ref: 
- Difference between `pipeline` and `pipe`: https://stackoverflow.com/questions/58875655/whats-the-difference-between-pipe-and-pipeline-on-streams